### ISSUE TITLE
metrics-server-deployment.yaml does not exist

### ISSUE THREAD

**Issue by asasaki4github** [takara9/vagrant-kubernetes#7]
When use 'vagrant up' to provision a master node to VBox on MacBook, task is failed as "metrics-server-deployment.yaml does not exist". Please tell me how to fix this error.

```
TASK [Install Metrics Server] **************************************************
changed: [master]

TASK [add a new line] **********************************************************
fatal: [master]: FAILED! => {"changed": false, "msg": "Path /home/vagrant/metrics-server/deploy/kubernetes/metrics-server-deployment.yaml does not exist !", "rc": 257}

PLAY RECAP *********************************************************************
master                     : ok=17   changed=13   unreachable=0    failed=1    skipped=3    rescued=0    ignored=0   

Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
```

**Comment by takara9** [comment]
Hello sasaki-san,

Please try this procedure and tell me outcome.

~~~
$ vagrant destroy -f
$ git pull
$ vagrant up
~~~

**Comment by asasaki4github** [comment]
I tried again, still same error.

```
asasaki@iMac vagrant-Kubernetes % vagrant destroy -f
==> master: Forcing shutdown of VM...
==> master: Destroying VM and associated drives...
==> node2: Forcing shutdown of VM...
==> node2: Destroying VM and associated drives...
==> node1: Forcing shutdown of VM...
==> node1: Destroying VM and associated drives...
asasaki@iMac vagrant-Kubernetes % git pull
Already up to date.
asasaki@iMac vagrant-Kubernetes % vagrant up
Bringing machine 'node1' up with 'virtualbox' provider...
Bringing machine 'node2' up with 'virtualbox' provider...
Bringing machine 'master' up with 'virtualbox' provider...
==> node1: Importing base box 'ubuntu/xenial64'...
==> node1: Matching MAC address for NAT networking...
==> node1: Checking if box 'ubuntu/xenial64' version '20200129.0.0' is up to date...
==> node1: A newer version of the box 'ubuntu/xenial64' for provider 'virtualbox' is
==> node1: available! You currently have version '20200129.0.0'. The latest is version
==> node1: '20200514.0.0'. Run `vagrant box update` to update.
==> node1: Setting the name of the VM: vagrant-kubernetes_node1_1590128706976_55099
==> node1: Clearing any previously set network interfaces...
==> node1: Preparing network interfaces based on configuration...
    node1: Adapter 1: nat
    node1: Adapter 2: hostonly
==> node1: Forwarding ports...
    node1: 22 (guest) => 2222 (host) (adapter 1)
==> node1: Running 'pre-boot' VM customizations...
==> node1: Booting VM...
==> node1: Waiting for machine to boot. This may take a few minutes...
    node1: SSH address: 127.0.0.1:2222
    node1: SSH username: vagrant
    node1: SSH auth method: private key
    node1: 
    node1: Vagrant insecure key detected. Vagrant will automatically replace
    node1: this with a newly generated keypair for better security.
    node1: 
    node1: Inserting generated public key within guest...
    node1: Removing insecure key from the guest if it's present...
    node1: Key inserted! Disconnecting and reconnecting using new SSH key...
==> node1: Machine booted and ready!
==> node1: Checking for guest additions in VM...
    node1: The guest additions on this VM do not match the installed version of
    node1: VirtualBox! In most cases this is fine, but in rare cases it can
    node1: prevent things such as shared folders from working properly. If you see
    node1: shared folder errors, please make sure the guest additions within the
    node1: virtual machine match the version of VirtualBox you have installed on
    node1: your host and reload your VM.
    node1: 
    node1: Guest Additions Version: 5.1.38
    node1: VirtualBox Version: 6.0
==> node1: Setting hostname...
==> node1: Configuring and enabling network interfaces...
==> node1: Mounting shared folders...
    node1: /vagrant => /Users/asasaki/Documents/GitHub/vagrant-kubernetes
==> node1: Running provisioner: ansible_local...
    node1: Installing Ansible...
Vagrant has automatically selected the compatibility mode '2.0'
according to the Ansible version installed (2.9.9).

Alternatively, the compatibility mode can be specified in your Vagrantfile:
https://www.vagrantup.com/docs/provisioning/ansible_common.html#compatibility_mode

    node1: Running ansible-playbook...

PLAY [all] *********************************************************************

TASK [Gathering Facts] *********************************************************
[DEPRECATION WARNING]: Distribution Ubuntu 16.04 on host node1 should use 
/usr/bin/python3, but is using /usr/bin/python for backward compatibility with 
prior Ansible releases. A future Ansible release will default to using the 
discovered platform python for this host. See https://docs.ansible.com/ansible/
2.9/reference_appendices/interpreter_discovery.html for more information. This 
feature will be removed in version 2.12. Deprecation warnings can be disabled 
by setting deprecation_warnings=False in ansible.cfg.
ok: [node1]

TASK [include vars] ************************************************************
ok: [node1]

TASK [debug] *******************************************************************
ok: [node1] => {
    "k8s_minor_ver": "14"
}

TASK [Add Docker GPG key] ******************************************************
changed: [node1]

TASK [Add Docker APT repository] ***********************************************
changed: [node1]

TASK [Install a list of packages] **********************************************
changed: [node1]

TASK [usermod -aG docker vagrant] **********************************************
changed: [node1]

TASK [Set sysctl] **************************************************************
changed: [node1]

TASK [Add GlusterFS Repository] ************************************************
changed: [node1]

TASK [Install GlusterFS client] ************************************************
changed: [node1]

TASK [add Kubernetes apt-key] **************************************************
changed: [node1]

TASK [add Kubernetes' APT repository] ******************************************
changed: [node1]

TASK [Install kubectl] *********************************************************
changed: [node1]

TASK [Install kubelet] *********************************************************
changed: [node1]

TASK [Install kubeadm] *********************************************************
changed: [node1]

TASK [include vars] ************************************************************
ok: [node1]

TASK [getting hostonly ip address] *********************************************
changed: [node1]

TASK [debug] *******************************************************************
ok: [node1] => {
    "ip.stdout_lines[1].split(':')[1].split(' ')[0]": "172.16.20.12"
}

TASK [change 10-kubeadm.conf for v1.11 or later] *******************************
changed: [node1]

TASK [change 10-kubeadm.conf for v1.9 or v1.10] ********************************
skipping: [node1]

TASK [daemon-reload and restart kubelet] ***************************************
changed: [node1]

PLAY RECAP *********************************************************************
node1                      : ok=20   changed=15   unreachable=0    failed=0    skipped=1    rescued=0    ignored=0   

==> node2: Importing base box 'ubuntu/xenial64'...
==> node2: Matching MAC address for NAT networking...
==> node2: Checking if box 'ubuntu/xenial64' version '20200129.0.0' is up to date...
==> node2: Setting the name of the VM: vagrant-kubernetes_node2_1590128826889_43081
==> node2: Fixed port collision for 22 => 2222. Now on port 2200.
==> node2: Clearing any previously set network interfaces...
==> node2: Preparing network interfaces based on configuration...
    node2: Adapter 1: nat
    node2: Adapter 2: hostonly
==> node2: Forwarding ports...
    node2: 22 (guest) => 2200 (host) (adapter 1)
==> node2: Running 'pre-boot' VM customizations...
==> node2: Booting VM...
==> node2: Waiting for machine to boot. This may take a few minutes...
    node2: SSH address: 127.0.0.1:2200
    node2: SSH username: vagrant
    node2: SSH auth method: private key
    node2: 
    node2: Vagrant insecure key detected. Vagrant will automatically replace
    node2: this with a newly generated keypair for better security.
    node2: 
    node2: Inserting generated public key within guest...
    node2: Removing insecure key from the guest if it's present...
    node2: Key inserted! Disconnecting and reconnecting using new SSH key...
==> node2: Machine booted and ready!
==> node2: Checking for guest additions in VM...
    node2: The guest additions on this VM do not match the installed version of
    node2: VirtualBox! In most cases this is fine, but in rare cases it can
    node2: prevent things such as shared folders from working properly. If you see
    node2: shared folder errors, please make sure the guest additions within the
    node2: virtual machine match the version of VirtualBox you have installed on
    node2: your host and reload your VM.
    node2: 
    node2: Guest Additions Version: 5.1.38
    node2: VirtualBox Version: 6.0
==> node2: Setting hostname...
==> node2: Configuring and enabling network interfaces...
==> node2: Mounting shared folders...
    node2: /vagrant => /Users/asasaki/Documents/GitHub/vagrant-kubernetes
==> node2: Running provisioner: ansible_local...
    node2: Installing Ansible...
Vagrant has automatically selected the compatibility mode '2.0'
according to the Ansible version installed (2.9.9).

Alternatively, the compatibility mode can be specified in your Vagrantfile:
https://www.vagrantup.com/docs/provisioning/ansible_common.html#compatibility_mode

    node2: Running ansible-playbook...

PLAY [all] *********************************************************************

TASK [Gathering Facts] *********************************************************
[DEPRECATION WARNING]: Distribution Ubuntu 16.04 on host node2 should use 
/usr/bin/python3, but is using /usr/bin/python for backward compatibility with 
prior Ansible releases. A future Ansible release will default to using the 
discovered platform python for this host. See https://docs.ansible.com/ansible/
2.9/reference_appendices/interpreter_discovery.html for more information. This 
feature will be removed in version 2.12. Deprecation warnings can be disabled 
by setting deprecation_warnings=False in ansible.cfg.
ok: [node2]

TASK [include vars] ************************************************************
ok: [node2]

TASK [debug] *******************************************************************
ok: [node2] => {
    "k8s_minor_ver": "14"
}

TASK [Add Docker GPG key] ******************************************************
changed: [node2]

TASK [Add Docker APT repository] ***********************************************
changed: [node2]

TASK [Install a list of packages] **********************************************
changed: [node2]

TASK [usermod -aG docker vagrant] **********************************************
changed: [node2]

TASK [Set sysctl] **************************************************************
changed: [node2]

TASK [Add GlusterFS Repository] ************************************************
changed: [node2]

TASK [Install GlusterFS client] ************************************************
changed: [node2]

TASK [add Kubernetes apt-key] **************************************************
changed: [node2]

TASK [add Kubernetes' APT repository] ******************************************
changed: [node2]

TASK [Install kubectl] *********************************************************
changed: [node2]

TASK [Install kubelet] *********************************************************
changed: [node2]

TASK [Install kubeadm] *********************************************************
changed: [node2]

TASK [include vars] ************************************************************
ok: [node2]

TASK [getting hostonly ip address] *********************************************
changed: [node2]

TASK [debug] *******************************************************************
ok: [node2] => {
    "ip.stdout_lines[1].split(':')[1].split(' ')[0]": "172.16.20.13"
}

TASK [change 10-kubeadm.conf for v1.11 or later] *******************************
changed: [node2]

TASK [change 10-kubeadm.conf for v1.9 or v1.10] ********************************
skipping: [node2]

TASK [daemon-reload and restart kubelet] ***************************************
changed: [node2]

PLAY RECAP *********************************************************************
node2                      : ok=20   changed=15   unreachable=0    failed=0    skipped=1    rescued=0    ignored=0   

==> master: Importing base box 'ubuntu/xenial64'...
==> master: Matching MAC address for NAT networking...
==> master: Checking if box 'ubuntu/xenial64' version '20200129.0.0' is up to date...
==> master: Setting the name of the VM: vagrant-kubernetes_master_1590128950426_53574
==> master: Fixed port collision for 22 => 2222. Now on port 2201.
==> master: Clearing any previously set network interfaces...
==> master: Preparing network interfaces based on configuration...
    master: Adapter 1: nat
    master: Adapter 2: hostonly
==> master: Forwarding ports...
    master: 22 (guest) => 2201 (host) (adapter 1)
==> master: Running 'pre-boot' VM customizations...
==> master: Booting VM...
==> master: Waiting for machine to boot. This may take a few minutes...
    master: SSH address: 127.0.0.1:2201
    master: SSH username: vagrant
    master: SSH auth method: private key
    master: 
    master: Vagrant insecure key detected. Vagrant will automatically replace
    master: this with a newly generated keypair for better security.
    master: 
    master: Inserting generated public key within guest...
    master: Removing insecure key from the guest if it's present...
    master: Key inserted! Disconnecting and reconnecting using new SSH key...
==> master: Machine booted and ready!
==> master: Checking for guest additions in VM...
    master: The guest additions on this VM do not match the installed version of
    master: VirtualBox! In most cases this is fine, but in rare cases it can
    master: prevent things such as shared folders from working properly. If you see
    master: shared folder errors, please make sure the guest additions within the
    master: virtual machine match the version of VirtualBox you have installed on
    master: your host and reload your VM.
    master: 
    master: Guest Additions Version: 5.1.38
    master: VirtualBox Version: 6.0
==> master: Setting hostname...
==> master: Configuring and enabling network interfaces...
==> master: Mounting shared folders...
    master: /vagrant => /Users/asasaki/Documents/GitHub/vagrant-kubernetes
==> master: Running provisioner: ansible_local...
    master: Installing Ansible...
Vagrant has automatically selected the compatibility mode '2.0'
according to the Ansible version installed (2.9.9).

Alternatively, the compatibility mode can be specified in your Vagrantfile:
https://www.vagrantup.com/docs/provisioning/ansible_common.html#compatibility_mode

    master: Running ansible-playbook...

PLAY [all] *********************************************************************

TASK [Gathering Facts] *********************************************************
[DEPRECATION WARNING]: Distribution Ubuntu 16.04 on host master should use 
/usr/bin/python3, but is using /usr/bin/python for backward compatibility with 
prior Ansible releases. A future Ansible release will default to using the 
discovered platform python for this host. See https://docs.ansible.com/ansible/
2.9/reference_appendices/interpreter_discovery.html for more information. This 
feature will be removed in version 2.12. Deprecation warnings can be disabled 
by setting deprecation_warnings=False in ansible.cfg.
ok: [master]

TASK [include vars] ************************************************************
ok: [master]

TASK [debug] *******************************************************************
ok: [master] => {
    "k8s_minor_ver": "14"
}

TASK [Add Docker GPG key] ******************************************************
changed: [master]

TASK [Add Docker APT repository] ***********************************************
changed: [master]

TASK [Install a list of packages] **********************************************
changed: [master]

TASK [usermod -aG docker vagrant] **********************************************
changed: [master]

TASK [Set sysctl] **************************************************************
changed: [master]

TASK [Add GlusterFS Repository] ************************************************
changed: [master]

TASK [Install GlusterFS client] ************************************************
changed: [master]

TASK [add Kubernetes apt-key] **************************************************
changed: [master]

TASK [add Kubernetes' APT repository] ******************************************
changed: [master]

TASK [Install kubectl] *********************************************************
changed: [master]

TASK [Install kubelet] *********************************************************
changed: [master]

TASK [Install kubeadm] *********************************************************
changed: [master]

TASK [include vars] ************************************************************
ok: [master]

TASK [getting hostonly ip address] *********************************************
changed: [master]

TASK [debug] *******************************************************************
ok: [master] => {
    "ip.stdout_lines[1].split(':')[1].split(' ')[0]": "172.16.20.11"
}

TASK [change 10-kubeadm.conf for v1.11 or later] *******************************
changed: [master]

TASK [change 10-kubeadm.conf for v1.9 or v1.10] ********************************
skipping: [master]

TASK [daemon-reload and restart kubelet] ***************************************
changed: [master]

PLAY RECAP *********************************************************************
master                     : ok=20   changed=15   unreachable=0    failed=0    skipped=1    rescued=0    ignored=0   

==> master: Running provisioner: ansible_local...
Vagrant has automatically selected the compatibility mode '2.0'
according to the Ansible version installed (2.9.9).

Alternatively, the compatibility mode can be specified in your Vagrantfile:
https://www.vagrantup.com/docs/provisioning/ansible_common.html#compatibility_mode

    master: Running ansible-playbook...

PLAY [master] ******************************************************************

TASK [Gathering Facts] *********************************************************
[DEPRECATION WARNING]: Distribution Ubuntu 16.04 on host master should use 
/usr/bin/python3, but is using /usr/bin/python for backward compatibility with 
prior Ansible releases. A future Ansible release will default to using the 
discovered platform python for this host. See https://docs.ansible.com/ansible/
2.9/reference_appendices/interpreter_discovery.html for more information. This 
feature will be removed in version 2.12. Deprecation warnings can be disabled 
by setting deprecation_warnings=False in ansible.cfg.
ok: [master]

TASK [include k8s vartsion] ****************************************************
ok: [master]

TASK [kubeadm reset v1.11 or later] ********************************************
changed: [master]

TASK [kubeadm reset v1.10 or eary] *********************************************
skipping: [master]

TASK [getting hostonly ip address] *********************************************
changed: [master]

TASK [kubeadm init] ************************************************************
changed: [master]

TASK [file] ********************************************************************
changed: [master]

TASK [kubeadm < 1.13] **********************************************************
skipping: [master]

TASK [kubeadm >= 1.14] *********************************************************
changed: [master]

TASK [change config.yaml] ******************************************************
ok: [master]

TASK [daemon-reload and restart kubelet] ***************************************
changed: [master]

TASK [mkdir kubeconfig] ********************************************************
changed: [master]

TASK [mkdir /vagrant/kubeconfig] ***********************************************
ok: [master]

TASK [chmod admin.conf] ********************************************************
changed: [master]

TASK [copy config to home dir] *************************************************
changed: [master]

TASK [download Manifest of Flannel for k8s ver <=11] ***************************
skipping: [master]

TASK [download Manifest of Flannel for k8s ver > 11] ***************************
changed: [master]

TASK [add a new line] **********************************************************
changed: [master]

TASK [Deploy Flannel] **********************************************************
changed: [master]

TASK [Install Metrics Server] **************************************************
changed: [master]

TASK [add a new line] **********************************************************
fatal: [master]: FAILED! => {"changed": false, "msg": "Path /home/vagrant/metrics-server/deploy/kubernetes/metrics-server-deployment.yaml does not exist !", "rc": 257}

PLAY RECAP *********************************************************************
master                     : ok=17   changed=13   unreachable=0    failed=1    skipped=3    rescued=0    ignored=0   

Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
```

**Comment by asasaki4github** [comment]
It seems only the master nodes is up.

```
asasaki@iMac vagrant-Kubernetes % vagrant ssh master
Welcome to Ubuntu 16.04.6 LTS (GNU/Linux 4.4.0-173-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage


79 packages can be updated.
56 updates are security updates.

New release '18.04.4 LTS' available.
Run 'do-release-upgrade' to upgrade to it.


vagrant@master:~$ kubectl get nodes
NAME     STATUS   ROLES    AGE     VERSION
master   Ready    master   7m17s   v1.14.10
```

**Comment by takara9** [comment]
It seems to be older version ansible code, but I don't know why it didn't update to latest code by "git pull". 

please see this point in below log  "... base box 'ubuntu/bionic64'..."
but your log indicates  "... base box 'ubuntu/xenial64'..."
it means Ubuntu early version.

~~~
maho:vagrant-kubernetes maho$ git pull
Already up to date.
maho:vagrant-kubernetes maho$ git branch
* master
maho:vagrant-kubernetes maho$ vagrant up
==> vagrant: A new version of Vagrant is available: 2.2.9 (installed version: 2.2.6)!
==> vagrant: To upgrade visit: https://www.vagrantup.com/downloads.html

Bringing machine 'master' up with 'virtualbox' provider...
Bringing machine 'node1' up with 'virtualbox' provider...
Bringing machine 'node2' up with 'virtualbox' provider...
==> master: Importing base box 'ubuntu/bionic64'...
==> master: Matching MAC address for NAT networking...
==> master: Checking if box 'ubuntu/bionic64' version '20191218.0.0' is up to date...
==> master: A newer version of the box 'ubuntu/bionic64' for provider 'virtualbox' is
==> master: available! You currently have version '20191218.0.0'. The latest is version
==> master: '20200521.0.0'. Run `vagrant box update` to update.
==> master: Setting the name of the VM: vagrant-kubernetes_master_1590126587314_74974
==> master: Clearing any previously set network interfaces...
==> master: Preparing network interfaces based on configuration...
    master: Adapter 1: nat
    master: Adapter 2: hostonly
==> master: Forwarding ports...
    master: 22 (guest) => 2222 (host) (adapter 1)
==> master: Running 'pre-boot' VM customizations...
==> master: Booting VM...
==> master: Waiting for machine to boot. This may take a few minutes...

...

TASK [node : daemon-reload and restart kubelet] ********************************
changed: [node2]

PLAY RECAP *********************************************************************
node2                      : ok=24   changed=18   unreachable=0    failed=0    skipped=11   rescued=0    ignored=0   

maho:vagrant-kubernetes maho$ vagrant ssh master1
The machine with the name 'master1' was not found configured for
this Vagrant environment.
maho:vagrant-kubernetes maho$ vagrant ssh master
Welcome to Ubuntu 18.04.3 LTS (GNU/Linux 4.15.0-72-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage

  System information as of Fri May 22 06:58:56 UTC 2020

  System load:  0.14              Users logged in:        0
  Usage of /:   32.7% of 9.63GB   IP address for enp0s3:  10.0.2.15
  Memory usage: 35%               IP address for enp0s8:  172.16.20.11
  Swap usage:   0%                IP address for docker0: 172.17.0.1
  Processes:    142               IP address for cni0:    10.244.0.1


129 packages can be updated.
77 updates are security updates.


vagrant@master:~$ kubectl get node -o wide
NAME     STATUS   ROLES    AGE     VERSION    INTERNAL-IP    EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION      CONTAINER-RUNTIME
master   Ready    master   7m40s   v1.14.10   172.16.20.11   <none>        Ubuntu 18.04.3 LTS   4.15.0-72-generic   docker://18.6.1
node1    Ready    <none>   3m57s   v1.14.10   172.16.20.12   <none>        Ubuntu 18.04.3 LTS   4.15.0-72-generic   docker://18.6.1
node2    Ready    <none>   27s     v1.14.10   172.16.20.13   <none>        Ubuntu 18.04.3 LTS   4.15.0-72-generic   docker://18.6.1
~~~


Normally "git pull" is effective to update latest code. but this time, it did not work.
To ensure get the latest code, please try again from git clone like below.

~~~
$ git clone https://github.com/takara9/vagrant-kubernetes k8s
$ cd k8s
$ vagarnt up
...
~~~

Thanks !
Maho

**Comment by asasaki4github** [comment]
Maho-san,

Git clone with the new folder 'k8s' and tried again with the latest code (ubuntu/bionic64), it works fine. Thanks for your kindly advice!
